# Introduction

This project was performed to identify whether there was a correlation between number of pages (num_pages), number of ratings (ratings_count), number of review (text_reviews_count) and the average rating (average_rating) of a book. Pandas was used in combination with statsmodels in order to perform the analysis and come to an conclusion.

In [1]:
import pandas as pd
import statsmodels.api as sm

In [3]:
books = pd.read_csv('Module 6-books.csv', index_col = 0)

In [8]:
books.head()
books.shape
#13719, 9

books.describe()

average_rating        isbn13     num_pages  ratings_count  \
count    13719.000000  1.371900e+04  13719.000000   1.371900e+04   
mean         3.930265  9.763320e+12    342.380713   1.775909e+04   
std          0.359432  3.986681e+11    252.618887   1.129371e+05   
min          0.000000  8.987060e+09      0.000000   0.000000e+00   
25%          3.770000  9.780000e+12    196.000000   8.250000e+01   
50%          3.960000  9.780000e+12    301.000000   6.300000e+02   
75%          4.130000  9.780000e+12    421.000000   4.737000e+03   
max          5.000000  9.790000e+12   6576.000000   5.629932e+06   

       text_reviews_count  
count        13719.000000  
mean           533.439901  
std           2528.565801  
min              0.000000  
25%              7.000000  
50%             40.000000  
75%            222.000000  
max          93619.000000

In [9]:
books['authors'].describe()

count               13719
unique               7605
top       Agatha Christie
freq                   69
Name: authors, dtype: object

In [11]:
a = books['average_rating']
n = books['num_pages']

model = sm.OLS(a, n).fit()
summary = model.summary()
summary

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:         average_rating   R-squared (uncentered):                   0.657
Model:                            OLS   Adj. R-squared (uncentered):              0.657
Method:                 Least Squares   F-statistic:                          2.625e+04
Date:                Thu, 03 Mar 2022   Prob (F-statistic):                        0.00
Time:                        19:46:44   Log-Likelihood:                         -30966.
No. Observations:               13719   AIC:                                  6.193e+04
Df Residuals:                   13718   BIC:                                  6.194e+04
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
num_pages      0.0075   4.64e-05    162.010      0.000       0.007       0.008
==============================================================================
Omnibus:                    12972.390   Durbin-Watson:                   1.021
Prob(Omnibus):                  0.000   Jarque-Bera (JB):          1709661.989
Skew:                          -4.173   Prob(JB):                         0.00
Kurtosis:                      57.049   Cond. No.                         1.00
==============================================================================

Notes:
[1] R² is computed without centering (uncentered) since the model does not contain a constant.
[2] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [14]:
a = books['average_rating']
r = books['ratings_count']

model = sm.OLS(a, r).fit()
summary = model.summary()
summary

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:         average_rating   R-squared (uncentered):                   0.025
Model:                            OLS   Adj. R-squared (uncentered):              0.025
Method:                 Least Squares   F-statistic:                              353.1
Date:                Thu, 03 Mar 2022   Prob (F-statistic):                    8.56e-78
Time:                        19:49:06   Log-Likelihood:                         -38127.
No. Observations:               13719   AIC:                                  7.626e+04
Df Residuals:                   13718   BIC:                                  7.626e+04
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
ratings_count  5.469e-06   2.91e-07     18.790      0.000     4.9e-06    6.04e-06
==============================================================================
Omnibus:                    26245.201   Durbin-Watson:                   0.057
Prob(Omnibus):                  0.000   Jarque-Bera (JB):        102418322.791
Skew:                         -14.586   Prob(JB):                         0.00
Kurtosis:                     425.279   Cond. No.                         1.00
==============================================================================

Notes:
[1] R² is computed without centering (uncentered) since the model does not contain a constant.
[2] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

In [16]:
a = books ['average_rating']
t = books['text_reviews_count']

model=sm.OLS(a, t).fit()
summary = model.summary()
summary

<class 'statsmodels.iolib.summary.Summary'>
"""
                                 OLS Regression Results                                
=======================================================================================
Dep. Variable:         average_rating   R-squared (uncentered):                   0.044
Model:                            OLS   Adj. R-squared (uncentered):              0.044
Method:                 Least Squares   F-statistic:                              625.6
Date:                Thu, 03 Mar 2022   Prob (F-statistic):                   4.67e-135
Time:                        19:50:11   Log-Likelihood:                         -37995.
No. Observations:               13719   AIC:                                  7.599e+04
Df Residuals:                   13718   BIC:                                  7.600e+04
Df Model:                           1                                                  
Covariance Type:            nonrobust                                                  
======================================================================================
                         coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------------
text_reviews_count     0.0003   1.28e-05     25.012      0.000       0.000       0.000
==============================================================================
Omnibus:                    24068.974   Durbin-Watson:                   0.093
Prob(Omnibus):                  0.000   Jarque-Bera (JB):         42128909.235
Skew:                         -12.302   Prob(JB):                         0.00
Kurtosis:                     273.361   Cond. No.                         1.00
==============================================================================

Notes:
[1] R² is computed without centering (uncentered) since the model does not contain a constant.
[2] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

## Summary

Based on data collected, it was determined that there are 13,719 total books in the document and of those, 7,605 are unique. The average rating for books is 3.93, the average number of pages is 342, the average ratings count is 17759.09, and average text reviews count is 533. The top author identified was Agatha Christie and appears in the list 69 times.
Based on the results from the OLS regression and examining the P-value of 0.000, we have failed to reject the null hypothesis that average ratings is independent of number of pages. There is a possible correlation between how many pages a book contains and the average rating the book receives. In addition, when examining the P-value (0.000) for the relationship between average rating of a book and number of ratings, it also suggests that there is a possible correlation between the average rating of a book and the number of ratings it receives. Additionally, the P-value (0.000) when examining the average rating of a book and total number of written text reviews, also suggests that there is a correlation between these two variables. 
Overall, it appears there is relationship between the average rating the book receives, the number of written text reviews, number of pages, and average rating. 
